reconcile/aus/base.py

```diff
@@ -305,26 +313,25 @@ def summarize(self, ocm_org_name: str) -> str:
 class ControlPlaneUpgradePolicy(AbstractUpgradePolicy):
     """Class to create and delete ControlPlanUpgradePolicies in OCM"""
 
-    def create(self, ocm: OCM) -> None:
+    def create(self, ocm_api: OCMBaseClient) -> None:
         policy = {
             "version": self.version,
             "schedule_type": "manual",
             "upgrade_type": "ControlPlane",
-            "cluster_id": ocm.cluster_ids[self.cluster],
+            "cluster_id": self.cluster.id,
             "next_run": self.next_run,
         }
-        ocm.create_control_plane_upgrade_policy(self.cluster, policy)
+        create_control_plane_upgrade_policy(ocm_api, self.cluster.id, policy)
 
-    def delete(self, ocm: OCM) -> None:
-        item = {
-            "id": self.id,
-        }
-        ocm.delete_control_plane_upgrade_policy(self.cluster, item)
+    def delete(self, ocm_api: OCMBaseClient) -> None:
+        if not self.id:
+            return
```

Comment:
isn't this a bug condition? Shouldn't this raise some alert? Like warn-level logging

Reply:
Applies for other deletes as well

Reply:
ack. i'm raising an exception now because a call to delete can not happen for an unsaved policy